src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -31,8 +32,8 @@ internal partial class Number
         private const int Int64Precision = 19;
         private const int UInt64Precision = 20;
 
-        /// <summary>256-element map from an ASCII char to its hex value, e.g. arr['b'] == 11. 0xFF means it's not a hex digit.</summary>
-        internal static readonly int[] s_charToHexLookup =
+        /// <summary>128-element map from an ASCII char to its hex value, e.g. arr['b'] == 11. 0xFF means it's not a hex digit.</summary>
+        internal static readonly byte[] s_charToHexLookup =
```

Comment:
This should be `ReadOnlySpan<byte>` property (look for "uses C# compiler's optimization for static byte[] data" in CoreLib for an example how to do it).

Reply:
When I changed the code in this way, it became noticeably slower. Maybe I compiling incorrectly? After a complete compilation of the new version, I run this script to benchmark small changes:

```
call build.cmd x64 release skiptests skipbuildpackages skipnative
call run.cmd build -Project="tests\build.proj" -BuildOS=Windows_NT -BuildType=Release -BuildArch=x64 -BatchRestorePackages
call tests\runtest.cmd x64 release generatelayoutonly
```